src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -226,6 +238,34 @@ public void refreshItemSetChoices() {
         }
     }
 
+    // Recalculates screen properties that are dependent on user input
+    public void refresh() {
```

Comment:
nit: naming suggestion. Typically `refresh` on it's own implies refreshing the full model but here we are only re-computing some dependent state. 
```suggestion
    public void refreshDependentState() {
```

Not a great name so feel free to adjust.